### PR TITLE
Changes to DevSite Changes for Workbox

### PR DESCRIPTION
**Pull request by gauntface**
Workbox Appengine changes

### CODE REVIEW

File: gulp-tasks/build.js [hunk 1]
```diff
@@ -20,6 +20,10 @@ gulp.task('build:announcement', function() {
   const globOpts = {
     srcBase: 'src/content/en/',
     prefixBase: true,
+    ignore: [
```
Comment (petele) [inline gulp-tasks/build.js]: Why are you disabling this? It won't be disabled on the real site.

File: gulp-tasks/claat-export.js [hunk 2]
```diff
@@ -4,7 +4,6 @@ const gulp = require('gulp');
 const path = require('path');
 const glob = require('globule');
 const wfHelper = require('./wfHelper');
-const runSequence = require('run-sequence');
```
Comment (petele) [inline gulp-tasks/claat-export.js]: Please rebase and add this stuff back

File: gulp-tasks/workbox:generate-contributors.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+const fs = require('fs-extra');
+const path = require('path');
+const gulp = require('gulp');
+
+const MAX_COLLABORATORS = 20;
+
+function getContributorHTML(contributorStats) {
+  const contributor = contributorStats.author;
+  var html = `
+<div class="wb-contrib">
```
Comment (petele) [inline gulp-tasks/workbox:generate-contributors.js]: can we do this using templates so we can keep code to build and template code separate?

File: gulp-tasks/workbox:generate-contributors.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
+const GitHubApi = require('github');
```
Comment (petele) [inline gulp-tasks/workbox:generate-contributors.js]: some filesystems may puke on a filename with a colon in it, can we change the colon?

File: gulp-tasks/workbox:generate-contributors.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  <a class="wb-contrib__link" href="${contributor.url}" target="_blank">
+    ${contributor.login}
+  </a>
+</div>
+`;
+  return html;
+}
+
+gulp.task(`workbox:generate-contributors`, function() {
+  if (!process.env.GITHUB_TOKEN) {
```
Comment (petele) [inline gulp-tasks/workbox:generate-contributors.js]: Let's change this to the same way that the youtube token works, that way not everyone needs to have this setup.

File: gulp-tasks/workbox:generate-contributors.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  })
+  .then((contributorStats) => {
+    contributorStats.sort((a, b) => b.total - a.total);
+    if (contributorStats.length > MAX_COLLABORATORS) {
+      contributorStats = contributorStats.splice(0, MAX_COLLABORATORS);
+    }
+    return contributorStats.map(getContributorHTML).join('');
+  })
+  .then((html) => {
+    const finalHtml = `{% framebox height="auto" %}
```
Comment (petele) [inline gulp-tasks/workbox:generate-contributors.js]: Same comment about templating.

File: gulp-tasks/workbox:reference-docs-v3.js [hunk 7]
```diff
@@ -0,0 +1,194 @@
+const gulp = require('gulp');
```
Comment (petele) [inline gulp-tasks/workbox:reference-docs-v3.js]: same comment about filename

File: gulp-tasks/workbox:generate-contributors.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    ${contributor.login}
+  </a>
+</div>
+`;
+  return html;
+}
+
+gulp.task(`workbox:generate-contributors`, function() {
+  if (!process.env.GITHUB_TOKEN) {
+    console.warn(`Unable to build Workbox contributors due to no ` +
```
Comment (petele) [inline gulp-tasks/workbox:generate-contributors.js]: to keep things consistent, can you please use `gutil.log` (and throughout)

File: gulp-tasks/workbox:reference-docs-v3.js [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+const remoteGitTags = require('remote-git-tags');
+const semver = require('semver');
+const fse = require('fs-extra');
+const path = require('path');
+const os = require('os');
+const spawn = require('child_process').spawn;
```
Comment (petele) [inline gulp-tasks/workbox:reference-docs-v3.js]: in wfHelper.js there's a `function promisedExec(cmd, cwd)` that will does all this for you.

File: gulp-tasks/build.js (hunk 1 above)
Comment (gauntface) [reply gulp-tasks/build.js]: Because I don't want the announcements on workbox.

File: gulp-tasks/workbox:generate-contributors.js (hunk 3 above)
Comment (gauntface) [reply gulp-tasks/workbox:generate-contributors.js]: Can you give me an example of doing this please - I struggle with figuring out the devsite stuff at the best of times.

File: gulp-tasks/build.js (hunk 1 above)
Comment (petele) [reply gulp-tasks/build.js]: Understood, but we want to be able to put them across all pages, so for now, let's leave them on and we can talk about removing them later.

File: gulp-tasks/workbox-generate-contributors.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+const MAX_COLLABORATORS = 20;
+
+/**
+ * Generates a list of contributors with links and images into
+ * tools/workbox/templates using the Github API to get the top 20
+ * contributors.
+ */
+
+gulp.task(`workbox-generate-contributors`, function() {
+  var token = process.env.GITHUB_TOKEN;
```
Comment (petele) [inline gulp-tasks/workbox-generate-contributors.js]: var, var? not let? what are we, in 1997? :P j/k

**Pull request merged**